R/descriptives.R

```diff
@@ -1881,3 +1948,339 @@ Descriptives <- function(jaspResults, dataset, options) {
+
+  if (options[["densityPlotSeparate"]] != "") {
+    separator <- .readDataSetToEnd(columns.as.factor = options[["densityPlotSeparate"]])
+    separator <- separator[,, drop = TRUE]
+  }
+
+  for (i in seq_along(variables)) {
+
+    if (!is.double(dataset[[i]])) next
+
+    variableName <- variables[[i]]
+    variable <- .readDataSetToEnd(columns.as.numeric = variableName)
+    variable <- variable[,, drop = TRUE]
+
+    if (options[["densityPlotSeparate"]] != "") {
+      data <- data.frame(variable, separator)
+    } else {
+      data <- data.frame(variable)
+    }
+
+    if (options[["splitby"]] != "") {
+      container[[variableName]] <- createJaspContainer(variableName)
+      splitBy <- .readDataSetToEnd(columns.as.factor = options[["splitby"]])
+      data$split <- splitBy[,, drop = TRUE]
+      data <- na.omit(data)
+      groups <- levels(data$split)
+
+      for (g in seq_along(groups)) {
+        active <- groups[g] == data$split
+        .descriptivesDensityPlotsFill(container[[variableName]], data[active, ], groups[g], variableName, g, options)
+      }
+    } else {
+      data <- na.omit(data)
+      .descriptivesDensityPlotsFill(container, data, variableName, variableName, i, options)
+    }
+  }
+}
+
+.descriptivesDensityPlotsFill <- function(container, data, plotName, axeName, position, options) {
+
+  data$split <- NULL
+  plotSize <- c(500, 500)
+  trans <- 1 - (options[["transparency"]]/100)
+  if (options[["densityPlotSeparate"]] != "") {
+    plotSize <- plotSize + c(300, 300)
+    p <- ggplot2::ggplot(data, ggplot2::aes(x = variable, fill = separator))
+    scale_fill <- jaspGraphs::scale_JASPfill_discrete(palette = options[["colorPalette"]],
+                                                      name = options[["densityPlotSeparate"]])
+  } else {
+    p <- ggplot2::ggplot(data, ggplot2::aes(x = variable, fill = factor(1)))
+    scale_fill <- jaspGraphs::scale_JASPfill_discrete(palette = options[["colorPalette"]])
+  }
+
+  densPlot <- createJaspPlot(title = plotName, width = plotSize[1], height = plotSize[2], position = position)
+  if (options[["densityPlotSeparate"]] != "" && any(table(data$separator) == 1)) {
+    densPlot$setError(gettext("Levels within variable require at least two or more data points!"))
+  } else {
+    p <- p + ggplot2::geom_density(alpha = trans) +
+      ggplot2::xlab(axeName) +
+      ggplot2::ylab(gettext("Density"))
+
+    # Determine range of axes to generate pretty breaks
+    yRange <- ggplot2::ggplot_build(p)$layout$panel_scales_y[[1]]$range$range
+    xRange <- ggplot2::ggplot_build(p)$layout$panel_scales_x[[1]]$range$range
+    yBreaks <- jaspGraphs::getPrettyAxisBreaks(yRange)
+    xBreaks <- jaspGraphs::getPrettyAxisBreaks(xRange)
+
+    p <- p + ggplot2::scale_y_continuous(breaks = yBreaks, limits = range(yBreaks)) +
+      ggplot2::scale_x_continuous(breaks = xBreaks, limits = range(xBreaks)) +
+      scale_fill +
+      jaspGraphs::geom_rangeframe(sides = "bl") +
```

Comment:
```suggestion
      jaspGraphs::geom_rangeframe() +
```
`sides = "bl"` is the default, right?

Reply:
yes, wrote it for clarification, but not required